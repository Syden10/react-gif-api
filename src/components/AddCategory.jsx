import { useState } from 'react';
import PropTypes from 'prop-types';
// import { useFetchGifs } from '../hooks/useFetchGifs';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    // const inputValue = inputValue.trim();
    if (inputValue.trim().length <= 1) return;
    // console.log(inputValue);
    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search gifs'
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
