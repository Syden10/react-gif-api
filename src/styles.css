* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
  box-sizing: border-box;
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 6px;
  border: 1px solid lightgray;
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

form {
  width: 100%;
}

button {
  appearance: none;
  border-radius: 19px;
  padding: 8px 16px;
  background: navy;
  color: white;
  border: none;
  font-size: 1rem;
  cursor: pointer;
  border: 2px solid navy;
}

button:hover {
  background: rgb(24, 0, 102);
  color: white;
}

.control-box {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.search-box {
  display: flex;
  width: 50%;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
  width: 23%;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}
