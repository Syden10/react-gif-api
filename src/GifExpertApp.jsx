import { useState } from 'react';
import { AddCategory, GifGrid } from './components/';

export default function GifExpertApp() {
  const [categories, setCategories] = useState(['cats']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  const clearCategories = () => {
    setCategories([]);
  };

  return (
    <>
      <h1>Gif App</h1>
      <div className='control-box'>
        <div className='search-box'>
          <AddCategory onNewCategory={onAddCategory} />
        </div>
        <div className='btn-box'>
          <button onClick={clearCategories}>Clear Gifs</button>
        </div>
      </div>
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
}
