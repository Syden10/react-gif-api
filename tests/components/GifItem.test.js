/** @jest-environment jsdom */
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('test on GifItem', () => {
  const title = 'A title';
  const url = 'https://test.com/image.jpg';
  test('should match snapshot', () => {
    const { container } = render(<GifItem url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show image and title', () => {
    render(<GifItem url={url} title={title} />);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
    // screen.debug();
    // console.log(screen.getByRole('img').src);
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
  });

  test('should render title', () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
