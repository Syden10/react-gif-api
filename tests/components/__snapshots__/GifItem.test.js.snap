// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A title"
      src="https://test.com/image.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
